### PR TITLE
add bijective base26 (excel like) x index in room

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2926 

image

### CODE REVIEW

File: inc/toolbox.class.php [hunk 1]
```diff
@@ -2859,4 +2859,19 @@ public static function useCache() {
          && (!defined('TU_USER') || defined('CACHED_TESTS'));
    }
 
+   /**
+    * Convert a integer index into an excel like alpha index (A, B, ..., AA, AB, ...)
```
Comment (trasher) [inline inc/toolbox.class.php]: Missing @since

File: js/common.js [hunk 2]
```diff
@@ -1043,6 +1043,21 @@ var replaceContent = function(editor, search, replace) {
    editor.setContent(body);
 };
 
+/**
+ * Convert a integer index into an excel like alpha index (A, B, ..., AA, AB, ...)
+ * @param  int index    the numeric index
```
Comment (trasher) [inline js/common.js]: Missing @since. s/int/integer/

File: tests/units/Toolbox.php [hunk 3]
```diff
@@ -120,4 +120,18 @@ public function testUncleanHtmlCrossSideScriptingDeepImageBase64() {
...
+      foreach ([
+         1   => 'A',
+         2   => 'B',
+         27  => 'AA',
+         28  => 'AB',
+         53  => 'BA',
+         702 => 'ZZ',
+         703 => 'AAA',
+      ] as $number => $bij_string) {
+         $this->string($bij_string)->isIdenticalTo(\Toolbox::getBijectiveIndex($number));
```
Comment (trasher) [inline tests/units/Toolbox.php]: That does not matter much, but string() is the string to test and isIdenticalTo() is the expected value :)

**Comment by orthagh** [comment]
fixed

File: js/common.js [hunk 4]
```diff
@@ -1043,6 +1043,22 @@ var replaceContent = function(editor, search, replace) {
    editor.setContent(body);
 };
 
+/**
+ * Convert a integer index into an excel like alpha index (A, B, ..., AA, AB, ...)
+ * @since  9.2
```
Comment (trasher) [inline js/common.js]: 9.3 :p

Comment (orthagh) [reply js/common.js]: my bad, fixed

**Pull request merged**